cylc/flow/data_store_mgr.py

```diff
@@ -2525,29 +2521,53 @@ def delta_task_ext_trigger(
         ext_trigger.time = update_time
         self.updates_pending = True
 
-    def delta_task_xtrigger(self, sig, satisfied):
-        """Create delta for change in task proxy xtrigger.
+    def _delta_xtrigger(
+        self, tp_id: str, label: str, sig: str, satisfied: bool,
+        t_update: float
+    ) -> None:
+        """Helper for the two xtrigger delta methods."""
+        tp_delta = self.updated[TASK_PROXIES].setdefault(
+            tp_id, PbTaskProxy(id=tp_id))
+        tp_delta.stamp = f'{tp_id}@{t_update}'
+        xtrigger = tp_delta.xtriggers[sig]
+        xtrigger.id = sig
+        xtrigger.label = label
+        xtrigger.satisfied = satisfied
+        xtrigger.time = t_update
+        self.updates_pending = True
```

Comment:
FYI, this does not work with subscriptions ATM as a result of https://github.com/cylc/cylc-flow/issues/6307

This issue is resolved on #6671, but requires a change of approach because whenever one xtrigger changes state, we need to list all xtriggers.

Reply:
OK, but no implications for this PR, right?

Reply:
Direct, no, but it would be good to build this atop the bugfix rather than in parallel. The bugfix is blocking https://github.com/cylc/cylc-ui/pull/2102 which would be handy for reviewing this if nothing else.